model/validate.go

```diff
@@ -72,7 +72,7 @@ func ValidateHost(hostId string, r *http.Request) (*host.Host, int, error) {
 		return nil, http.StatusBadRequest, errors.Errorf("Missing host secret for host %s", h.Id)
 	}
 	if secret != h.Secret {
-		return nil, http.StatusConflict, errors.Errorf("Invalid host secret for host %s", h.Id)
+		return nil, http.StatusUnauthorized, errors.Errorf("Invalid host secret for host %s", h.Id)
 	}
```

Comment:
why on earth is this in the model package? 

this also seems wrongish?

Reply:
> why on earth is this in the model package?

It it depended on by both the service and the rest packages. I'm open to moving it elsewhere, but the places that it can be moved to without a dependency cycle are limited. model is awkward but not crazy, as we are validating aspects of a business object.